agent/consul/subscribe_backend_test.go

```diff
@@ -204,11 +205,10 @@ func TestSubscribeBackend_IntegrationWithServer_TLSReload(t *testing.T) {
 	server.tlsConfigurator.Update(newConf)
 
 	// Try the subscribe call again
-	retryFailedConn(t, conn)
-
-	streamClient = pbsubscribe.NewStateChangeSubscriptionClient(conn)
-	_, err = streamClient.Subscribe(ctx, req)
-	require.NoError(t, err)
+	retry.Run(t, func(r *retry.R) {
+		_, err = streamClient.Subscribe(ctx, req)
+		require.NoError(r, err)
+	})
```

Comment:
There's a slight timing issue (<1s) that flakes the second try. I'm unsure why the previous code did not run into the same issue but I didn't consider this to be a significant regression.